test/support/helpers/scenarios.js

```diff
@@ -83,20 +79,34 @@ if (module.exports.scenarios.length === 0) {
   )
 }
 
+const disabledScenario = scenario =>
+  typeof scenario.disabled === 'string' && scenario.disabled
+
+/**
+ * For historical reasons, `disabledScenarioTest()` may receive test names
+ * with file extension. Although it may be useless in many cases, stripping
+ * them will make sure code always works.
+ */
+const disabledScenarioTest = (scenario, testName) =>
+  disabledScenario(scenario) ||
+  _.get(scenario, ['disabled', testName.replace(path.extname(testName), '')])
+
+module.exports.disabledScenarioTest = disabledScenarioTest
+
 module.exports.loadFSEventFiles = scenario => {
   const eventFiles = glob.sync(
     path.join(path.dirname(scenario.path), 'local', '*.json*')
   )
   const disabledEventsFile = name => {
     if (process.platform === 'win32' && name.indexOf('win32') === -1) {
       return 'darwin/linux test'
-    } else if (name.endsWith(disabledExtension)) {
-      return 'disabled case'
+    } else {
+      return disabledScenarioTest(scenario, `local/${name}`)
     }
   }
   return eventFiles.map(f => {
     const name = path.basename(f)
-    const disabled = scenario.disabled || disabledEventsFile(name)
+    const disabled = disabledScenario(scenario) || disabledEventsFile(name)
```

Comment:
It seems redundant to call `disabledScenario(scenario)` here since it will be called by `disabledScenarioTest()` in `disabledEventsFile()`.

Reply:
Right, but it won't be called first in `disabledEventsFile()`.
Which means the reason why it is disabled won't be the same.
I didn't wanted to change behavior here.
But all those `disabled*()` functions will definitely need refactoring.